### PR TITLE
NOISSUE - Move gRPC to api package

### PR DESCRIPTION
**Pull request by SammyOina**
# What type of PR is this?
this is a refactor of grpc api

## What does this do?
moves grpc to pkg from internal

## Which issue(s) does this PR fix/relate to?

- Related Issue #
- Resolves #

### CODE REVIEW

**Review by dborovcanin — CHANGES_REQUESTED** [review]
@SammyOina Please also update protoc and Go plugin:

```
-//   protoc-gen-go v1.34.2
-//   protoc        v5.28.3
+//   protoc-gen-go v1.35.2
+//   protoc        v5.29.0
```

File: api/grpc/auth/v1/auth.pb.go [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.35.2
-// 	protoc        v5.29.0
+// 	protoc-gen-go v1.34.2
+// 	protoc        v5.28.3
```
Comment (arvindh123) [inline api/grpc/auth/v1/auth.pb.go]: Please upgrade the `protoc-gen-go` version to `v1.35.2`  and `protoc` version to `v5.29.0`

**Review by arvindh123 — CHANGES_REQUESTED** [review]

File: tools/config/mockery.yaml [hunk 2]
```diff
@@ -6,35 +6,35 @@ filename: "{{.InterfaceName}}.go"
...
+  github.com/absmach/supermq/pkg/grpc/channels/v1:
     interfaces:
       ChannelsServiceClient:
         config:
           dir: "./channels/mocks"
           mockname: "ChannelsServiceClient"
           filename: "channels_client.go"
-  github.com/absmach/supermq/internal/grpc/groups/v1:
+  github.com/absmach/supermq/pkg/grpc/groups/v1:
     interfaces:
```
Comment (arvindh123) [inline tools/config/mockery.yaml:38]: Please change ```github.com/absmach/supermq/pkg/grpc/groups/v1``` 
to 

```go
  github.com/absmach/supermq/api/grpc/groups/v1:
```

Same applies for similar things in this same file

File: tools/config/mockery.yaml [hunk 3]
```diff
@@ -6,35 +6,35 @@ filename: "{{.InterfaceName}}.go"
...
+  github.com/absmach/supermq/pkg/grpc/token/v1:
     interfaces:
       TokenServiceClient:
         config:
           dir: "./auth/mocks"
           mockname: "TokenServiceClient"
           filename: "token_client.go"
-  github.com/absmach/supermq/internal/grpc/channels/v1:
+  github.com/absmach/supermq/pkg/grpc/channels/v1:
     interfaces:
```
Comment (arvindh123) [inline tools/config/mockery.yaml:31]:
```suggestion
  github.com/absmach/supermq/api/grpc/channels/v1:
```

File: tools/config/mockery.yaml [hunk 4]
```diff
@@ -6,35 +6,35 @@ filename: "{{.InterfaceName}}.go"
 outpkg: "mocks"
 boilerplate-file: "./tools/config/boilerplate.txt"
 packages:
-  github.com/absmach/supermq/internal/grpc/clients/v1:
+  github.com/absmach/supermq/pkg/grpc/clients/v1:
     interfaces:
```
Comment (arvindh123) [inline tools/config/mockery.yaml:10]:
```suggestion
  github.com/absmach/supermq/api/grpc/clients/v1:
```

File: tools/config/mockery.yaml [hunk 5]
```diff
@@ -6,35 +6,35 @@ filename: "{{.InterfaceName}}.go"
...
       ClientsServiceClient:
         config:
           dir: "./clients/mocks"
           mockname: "ClientsServiceClient"
           filename: "clients_client.go"
-  github.com/absmach/supermq/internal/grpc/domains/v1:
+  github.com/absmach/supermq/pkg/grpc/domains/v1:
     interfaces:
```
Comment (arvindh123) [inline tools/config/mockery.yaml:17]:
```suggestion
  github.com/absmach/supermq/api/grpc/domains/v1:
```

File: tools/config/mockery.yaml [hunk 6]
```diff
@@ -6,35 +6,35 @@ filename: "{{.InterfaceName}}.go"
...
       DomainsServiceClient:
         config:
           dir: "./domains/mocks"
           mockname: "DomainsServiceClient"
           filename: "domains_client.go"
-  github.com/absmach/supermq/internal/grpc/token/v1:
+  github.com/absmach/supermq/pkg/grpc/token/v1:
     interfaces:
```
Comment (arvindh123) [inline tools/config/mockery.yaml:24]:
```suggestion
  github.com/absmach/supermq/api/grpc/token/v1:
```

**Review by arvindh123 — CHANGES_REQUESTED** [review]

**Review by arvindh123 — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**